storage.go

```diff
@@ -48,10 +48,30 @@ type Storage interface {
 
 	// InitialState returns the saved HardState and ConfState information.
 	InitialState() (pb.HardState, pb.ConfState, error)
-	// Entries returns a slice of log entries in the range [lo,hi).
-	// MaxSize limits the total size of the log entries returned, but
-	// Entries returns at least one entry if any.
+
+	// Entries returns a slice of log entries in the range [lo, hi). The maxSize
+	// limits the total size of the log entries returned, but Entries returns at
+	// least one entry if any.
+	//
+	// Entries that storage returns via this method are not modified by the raft
```

Comment:
It would help to be more opinionated here. As a user, I don't want various options on how it could work (at least not when it's complicated like that) but just be told what to do.
We likely want to say that the caller of `Entries` owns the slice and must not mutate it (and that `raft` adheres to that rule). We should then say something similar in the `Ready` struct, i.e. the application must not modify the slices handed to it by raft. According to the above it would be up to `raft` to not append into the slice when there's spare capacity.

Reply:
Reworded, PTAL.

`Ready` already states that all fields are read-only, so we're good there w.r.t. the entries slices too.